src/validator/FetchVerticesValidator.cpp

```diff
@@ -106,126 +108,133 @@ Status FetchVerticesValidator::prepareVertices() {
-                if (*expr->sym() != tagName_) {
-                    return Status::Error("Mismatched tag name");
-                }
-                // Check is prop name in schema
-                if (schema_->getFieldIndex(*expr->prop()) < 0) {
-                    LOG(ERROR) << "Unknown column `" << *expr->prop() << "' in tag `"
-                                << tagName_ << "'.";
-                    return Status::Error("Unknown column `%s' in tag `%s'.",
-                                            expr->prop()->c_str(),
-                                            tagName_.c_str());
-                }
-                propsName.emplace_back(*expr->prop());
-                gvColNames_.emplace_back(*expr->sym() + "." + *expr->prop());
-            }
-            colNames_.emplace_back(deduceColName(col));
-            auto typeResult = deduceExprType(col->expr());
-            NG_RETURN_IF_ERROR(typeResult);
-            outputs_.emplace_back(colNames_.back(), typeResult.value());
-            // TODO(shylock) think about the push-down expr
-        }
-        prop.set_props(std::move(propsName));
+        // empty for all
         props_.emplace_back(std::move(prop));
-
-        // insert the reserved properties expression be compatible with 1.0
-        // TODO(shylock) select kVid from storage
-        newYieldColumns_ = qctx_->objPool()->add(new YieldColumns());
-        // note eval vid by input expression
-        newYieldColumns_->addColumn(
-            new YieldColumn(new InputPropertyExpression(new std::string(VertexID)),
-                            new std::string(VertexID)));
-        for (auto col : yield->columns()) {
-            newYieldColumns_->addColumn(col->clone().release());
+        outputs_.emplace_back(VertexID, Value::Type::STRING);
+        colNames_.emplace_back(VertexID);
+        gvColNames_.emplace_back(VertexID);  // keep compatible with 1.0
+        for (std::size_t i = 0; i < schema_->getNumFields(); ++i) {
+            outputs_.emplace_back(schema_->getFieldName(i),
+                                    SchemaUtil::propTypeToValueType(schema_->getFieldType(i)));
+            colNames_.emplace_back(tagName_ + '.' + schema_->getFieldName(i));
+            gvColNames_.emplace_back(colNames_.back());
+        }
+    } else {
+        // all schema properties
+        const auto allTagsResult = qctx_->schemaMng()->getAllVerTagSchema(spaceId_);
+        NG_RETURN_IF_ERROR(allTagsResult);
+        const auto allTags = std::move(allTagsResult).value();
+        std::vector<std::pair<TagID, std::shared_ptr<const meta::NebulaSchemaProvider>>>
+            allTagsSchema;
+        allTagsSchema.reserve(allTags.size());
+        for (const auto &tag : allTags) {
+            allTagsSchema.emplace_back(tag.first, tag.second.back());
+        }
+        std::sort(allTagsSchema.begin(), allTagsSchema.end(), [](const auto &a, const auto &b) {
```

Comment:
Maybe you want to output order the same as creating schemas. but I don't think it's better to sort every time. you can make the cache is ordered.

Reply:
The schema will add/drop.